src/Collections/ArrayGeneric.php

```diff
@@ -107,7 +107,19 @@ public function valid(): bool
     public function __debugInfo(): array
     {
         return [
-            'data' => $this->data
+            'data' => $this->data,
         ];
     }
+
+    /**
+     * @param callable $filter
+     *
+     * @return $this
+     */
+    public function filter(callable $filter)
```

Comment:
I didn't want to do that because "normal" arrays don't have method, there are functions to work with arrays.

Reply:
Yeah fair enough, unfortunately though PHP array methods don't work on ArrayInterface instances, so this approach is basically required. I am tempted to fork this library though and develop it into more of a typed collection class rather than a plain ArrayInterface